### PR TITLE
fix: fix the wrong path when request dirname is same as root

### LINKED ISSUE #311: RootsPlugin causes watchpack watches a set of wrong paths
https://github.com/webpack/enhanced-resolve/blob/d1a3cb65cdf90754871a263172006e2902a2f86e/lib/RootsPlugin.js#L42

This gives a path `'/Users/wood/Desktop/webpack-bug/Users/wood/Desktop/webpack-bug/examples/relay/src/index.web'` See the following screenshot

image

because of that, watchpack watches a set of wrong paths

https://github.com/webpack/watchpack/blob/c651129195d84cc80e547a45be38861b5bbada1a/lib/watchEventSource.js#L56

image

Is there a reason of coding this `resolver.join(root, req.slice(1));`? 

Here is the reproducible repo https://github.com/wood1986/rootplugin-bug
1. yarn
2. yarn start

### PR DESCRIPTION
**Pull request by wood1986**
fix #311

### CODE REVIEW

File: lib/RootsPlugin.js [hunk 1]
```diff
@@ -39,7 +39,10 @@ class RootsPlugin {
 				forEachBail(
 					this.roots,
 					(root, callback) => {
-						const path = resolver.join(root, req.slice(1));
+						const path = root.includes(req)
```
Comment (alexander-akait) [inline lib/RootsPlugin.js:42]: What about `/foo/bar/foo/bar`?

Comment (wood1986) [reply lib/RootsPlugin.js]: it should be startsWith

**Comment by wood1986** [comment]
@alexander-akait Could you please review again?

**Review by alexander-akait — COMMENTED** [review]
Again, you can have `/foo/bar/foo/bar`, it is possible, fix is invalid, we should check `/foo/bar/` and `/foo/bar/foo/bar`

**Pull request closed without merging**